Include/internal/pycore_signal.h

```diff
@@ -29,6 +42,77 @@ extern "C" {
 #  define Py_NSIG 64               // Use a reasonable default value
 #endif
 
+#if defined(MS_WINDOWS) && defined(SOCKET)
+#  define INVALID_FD ((SOCKET)-1)
+#else
+#  define INVALID_FD (-1)
+#endif
```

Comment:
I'm more confident that falling back to plain `-1` here is okay.

Reply:
You'll probably need the cast to avoid getting warnings in places where it's used, so I think this change is correct right now.